### DISCUSSION TITLE
Broken image results

### DISCUSSION THREAD

**Discussion by EricDriussi** [searxng/searxng#1368] [Q&A]
So I followed [these](https://docs.searxng.org/admin/installation.html#installation-scripts) and [these](https://docs.searxng.org/admin/installation-nginx.html#installation-nginx) instructions and it all works really well with no issues except for the fact that the image and video results are broken (nginx returns a 403). So for example:

![2022-06-25_17-31-42](https://user-images.githubusercontent.com/46979145/175782586-2062c246-3b6d-408d-ad22-44e75e093428.png)

I can't really see anything off in the nginx logs, nor do I catch any complaints during installation.

The one difference is that it's not under `domain.com/searx` but rather under `searx.domain.com`.
I modified the `nginx` conf:

```
server {
    listen 80;
    listen [::]:80;

    server_name searx.domain.com;

    location / {
            proxy_pass         http://127.0.0.1:4004/;
                
            proxy_set_header   Host             $host;
            proxy_set_header   Connection       $http_connection;
            proxy_set_header   X-Real-IP        $remote_addr;
            proxy_set_header   X-Forwarded-For  $proxy_add_x_forwarded_for;
            proxy_set_header   X-Scheme         $scheme;
            proxy_set_header   X-Script-Name    /searx;
    }
        
    location /searx/static {
            alias /usr/local/searx/searx-src/searx/static;
    }
        
    location /morty {
            proxy_pass         http://127.0.0.1:3000/;
               
            proxy_set_header   Host             $host;
            proxy_set_header   Connection       $http_connection;
            proxy_set_header   X-Real-IP        $remote_addr;
            proxy_set_header   X-Forwarded-For  $proxy_add_x_forwarded_for;
            proxy_set_header   X-Scheme         $scheme;
    }
}
   

```

Relevant `settings.yml`
```
result_proxy:
    url : https://searx.domain.com/morty
    key : !!binary AAAAA
```

Hostname of course is `domain.com`.
I'm not sure what I'm missing.

Mind you, this is hosted on the cheapest VPS Vultr has to offer, sharing it with a nextcloud and vaultwarden instance.
So that might be an issue.

Oh, in case it's of any help, here's my `nginx.conf`:
```
user www-data;
worker_processes auto;
pid /run/nginx.pid;
include /etc/nginx/modules-enabled/*.conf;

events {
	worker_connections 768;
}

http {
	client_max_body_size 100M;

	sendfile on;
	tcp_nopush on;
	tcp_nodelay on;
	keepalive_timeout 65;
	types_hash_max_size 2048;

	include /etc/nginx/mime.types;
	default_type application/octet-stream;

	ssl_prefer_server_ciphers on;

	access_log /var/log/nginx/access.log;
	error_log /var/log/nginx/error.log;

	gzip on;
        gzip_vary on;
        gzip_min_length 10240;
        gzip_proxied expired no-cache no-store private auth;
        gzip_types text/plain text/css text/xml text/javascript application/x-javascript application/xml application/json;
        gzip_disable "MSIE [1-6]\.";

	include /etc/nginx/conf.d/*.conf;
	include /etc/nginx/sites-enabled/*;
}
```

Anyway, thanks in advance!

**Comment by ononoki1** [comment]
Can you directly access MortyProxy to see its full response (instead of just the error code 403)? Just right-click the request and open it in new tab.

---

BTW, if you just need to proxy images, you can use SearXNG's built-in `image_proxy` function instead of morty. Morty is only needed when you proxy web pages as well (like [this instance](https://search.ononoki.org)).

**Reply by EricDriussi** [reply]
Oh sure, here it is:
![image](https://user-images.githubusercontent.com/46979145/175785445-b137d472-85ba-491b-9e88-0ddcbc3eb84f.png)

It would be nice to have Morty working properly but I'll definitely use the built-in function for the time being, thanks for that!

**Reply by ononoki1** [reply]
You can try using a subdomain for morty instead of path `/morty`, e.g. *morty.domain.com* or *morty.searx.domain.com*.
